avalanche/benchmarks/scenarios/continuous_task_agnostic_scenario.py

```diff
@@ -0,0 +1,330 @@
+
+    def _reset_indices(self):
+        self.indices = torch.arange(self.n_samples).tolist()
+        if self.shuffle:
+            self.indices = torch.randperm(self.n_samples,
+                                          generator=self.rng).tolist()
+
+    def __iter__(self):
+        while True:
+            for idx in self.indices:
+                yield idx
+            self._reset_indices()
+
+    def __len__(self):
+        return self.n_samples
+
+
+def create_sub_exp_from_multi_exps(
+    online_benchmark: "OnlineCLScenario",
+    original_stream: Iterable[DatasetExperience[TClassificationDataset]],
+    samplers: Iterable[CyclicSampler],
+    exp_per_sample_list: Iterable[torch.Tensor],
+    sub_exp_id: int,
+    total_iters: int,
+    is_first_sub_exp: bool = False,
+    is_last_sub_exp: bool = False,
+) -> DatasetExperience[TClassificationDataset]:
+    """
+    Creates a sub-experience from a list of experiences.
+
+    :param online_benchmark: The online benchmark.
+    :param original_stream: The original stream.
+    :param samplers: A list of samplers, one for each experience in the
+            original stream.
+    :param exp_per_sample_list: A list of experience ids, one for each sample
+            in the sub-experience.
+    :param sub_exp_id: The id of the sub-experience.
+    :param total_iters: The total number of iterations.
+    :param is_first_sub_exp: Whether this is the first sub-experience.
+    :param is_last_sub_exp: Whether this is the last sub-experience.
+
+    :return: A sub-experience.
+    """
+
+    # Create sub-sets from each experience's dataset
+    all_subsets = []
+    n_samples_from_each_exp = [0 for _ in range(len(samplers))]
+
+    for exp_id in exp_per_sample_list.unique():
+        n_samples = sum(exp_per_sample_list == exp_id.item()).item()
+        n_samples_from_each_exp[exp_id.item()] += n_samples
+        rnd_indices = [next(samplers[exp_id]) for _ in range(n_samples)]
+        subset_i = classification_subset(
+            original_stream[exp_id.item()].dataset, rnd_indices)
+        all_subsets.append(subset_i)
+
+    # Concatenate all sub-sets
+    sub_exp_subset = concat_classification_datasets(all_subsets)
+    experience_size = len(sub_exp_subset)
+
+    exp = OnlineClassificationExperience(
+        current_experience=sub_exp_id,
+        origin_stream=None,  # type: ignore
+        benchmark=online_benchmark,
+        dataset=sub_exp_subset,
+        origin_experience=None,  # experience,
+        classes_in_this_experience=list(set(sub_exp_subset.targets)),
+        subexp_size=experience_size,
+        is_first_subexp=is_first_sub_exp,
+        is_last_subexp=is_last_sub_exp,
+        sub_stream_length=total_iters,
+        access_task_boundaries=False,
+    )
+
+    # For visualization purposes only
+    exp.n_samples_from_each_exp = n_samples_from_each_exp
+
+    return exp
+
+
+def _fixed_size_linear_decay_stream(
+    online_benchmark: "OnlineCLScenario",
```

Comment:
it's taking a benchmark but returning a stream? I think it would be more intuitive if it took a benchmark and returned a new benchmark. Also, why is this an `online_benchmark`? I think it should take large batches and split them.